Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/VaccinationRepository.kt

```diff
@@ -339,33 +339,27 @@ class VaccinationRepository @Inject constructor(
         }
     }
 
-    suspend fun updateBoosterRule(
-        personIdentifier: CertificatePersonIdentifier,
-        rule: DccValidationRule?
+    // removes booster rule information if newRuleId is null
+    suspend fun clearBoosterRuleInfo(
+        personIdentifier: CertificatePersonIdentifier
     ) {
         Timber.tag(TAG)
-            .d("updateBoosterRule(personIdentifier=%s, ruleIdentifier=%s)", personIdentifier, rule?.identifier)
+            .d("clearBoosterRuleInfo(personIdentifier=%s)", personIdentifier)
+
         internalData.updateBlocking {
+
             val vaccinatedPerson = singleOrNull { it.identifier == personIdentifier }
 
             if (vaccinatedPerson == null) {
-                Timber.tag(TAG).w("updateBoosterRule couldn't find person %s", personIdentifier.codeSHA256)
+                Timber.tag(TAG).w("clearBoosterRuleInfo couldn't find person %s", personIdentifier.codeSHA256)
                 return@updateBlocking this
             }
 
-            val data = when (rule) {
-                // Clear notification badge if user is not eligible for booster anymore
-                null -> vaccinatedPerson.data.copy(
-                    boosterRuleIdentifier = null,
-                    lastSeenBoosterRuleIdentifier = null
-                )
-                else -> vaccinatedPerson.data.copy(
-                    boosterRuleIdentifier = rule.identifier
-                )
-            }
+            val data =
+                vaccinatedPerson.data.copy(boosterRuleIdentifier = null, lastSeenBoosterRuleIdentifier = null)
```

Comment:
You need to remove `boosterRulesRepository` as dependency from here too 
https://github.com/corona-warn-app/cwa-app-android/blob/a1ea19fdc396ad7b4c268870e3dd42c2c087cbf8/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/VaccinationRepository.kt#L52

and all logic related to it , `DccWalletInfo` is providing such information now ;)

Reply:
removed